py3status/modules/getjson.py

```diff
@@ -68,28 +65,17 @@ class Py3status:
     def getjson(self):
         """
         """
-        response = {
-            'cached_until': self.py3.time_in(self.cache_timeout),
-        }
-
         try:
-            resp = self.py3.request(self.url, timeout=self.timeout)
-            status = resp.status_code == 200
-            resp = resp.json()
+            getjson = self.py3.request(self.url, timeout=self.timeout).json()
+            getjson = self.py3.flatten_dict(
+                getjson, delimiter=self.delimiter, intermediates=True)
         except self.py3.RequestException:
-            resp = None
-            status = False
-
-        if status:
-            response['full_text'] = self.py3.safe_format(
-                self.format,
-                self.py3.flatten_dict(
-                    resp, delimiter=self.delimiter, intermediates=True
-                )
-            )
-        else:
-            response['full_text'] = ''
-        return response
+            getjson = None
```

Comment:
> I think it's good to display GETJSON: {moo} 1232.34... Shows us that we need to fix {moo} placeholder somehow.... than GETJSON: 1234.43 then people might not know about this broken/wrong input/typos placeholders. 

Yeah this is important and we do this when we have data returned

> How do you suggest we '' the unknown placeholders... but still display the format? 

piss easy - the question is what is the best way to do this.

options:

* create our dict of all placeholders being `''`

* tell formatter we don't want empty ones shown.

* just decide that if we have no data that we don't show (eg `safe_format(format, None)` as opposed to `safe_format(format, {})` (here we show missing)

anyhow try to stick to single issues, so what behaviour we want, not other format ideas, otherwise it makes these discussions really hard to follow and we lose stuff.